core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddLocalExchangesForTaskScaleWriters.java

```diff
@@ -0,0 +1,126 @@
+    {
+        return MockConnectorFactory.builder()
+                .withSupportsReportingWrittenBytes(supportsWrittenBytes)
+                .withGetTableHandle(((session, schemaTableName) -> null))
+                .withName(name)
+                .build();
+    }
+
+    @Test
+    public void testLocalScaledWriterDistributionWithSupportsReportingWrittenBytes()
+    {
+        // when task scale writers is enabled
+        assertDistributedPlan(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+
+        // when task scale writers is disabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "false")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, GATHER, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+    }
+
+    @Test
+    public void testLocalScaledWriterDistributionWithoutSupportsReportingWrittenBytes()
+    {
+        // when task scale writers is enabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_dont_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+
+        // when task scale writers is disabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_dont_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "false")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
```

Comment:
why not `GATHER`?

Reply:
No, it has to be `REPARTITION` right? Because we are using `SCALED_WRITER_DISTRIBUTION` and not `SINGLE_DISTRIBUTION`

Reply:
I mean here `TASK_SCALE_WRITERS_ENABLED` is disabled, so why there is local exchange with `SCALED_WRITER_DISTRIBUTION` here (with `REPARTITION` type)??